cmd/cluster-init/main.go

```diff
@@ -141,13 +152,17 @@ func main() {
 
 	// Each step in the process is allowed to fail independently so that the diffs for the others can still be generated
 	errorCount := 0
-	for _, step := range []func(options) error{
-		updateJobs,
-		initClusterBuildFarmDir,
-		updateCiSecretBootstrap,
-		updateSecretGenerator,
-		updateSanitizeProwJobs,
-	} {
+	steps := []func(options) error{updateJobs}
+	if !o.update {
+		steps = append(steps,
+			initClusterBuildFarmDir,
+			updateCiSecretBootstrap,
```

Comment:
What would it take to make this work on update? What if we change the definition of what the secrets should look like? It's useful to update everything.

Reply:
I thought about that as well...Since we don't have the equivalent to a `jc.WriteToDir` in place for `secretbootstrap` I think it would result in a fair bit of effort. We would either need to go through every stanza in the file checking for the current cluster's configurations and remove them prior to doing a regenerate (this may not end up being too crazy, but there are a lot of configs in there), or we would need to write something like `prune` to prune the old configs when new ones are generated. Either way, I came to the conclusion that this might be more work than it would end up saving. Thoughts?